### ISSUE TITLE
Multiline statements

### ISSUE THREAD

**Issue by circle-gon** [TodePond/DreamBerd#598]
## multiline statements

Currently, the spec doesn't say much about multiline statements/expressions.

Are exclamations/questions supposed to end a statement? If so, then all statements can only be one line, because of AEMI.

```js
// 5 and 5, not 10
5 // seen as the literal 5
+ 5 // seen as a unary plus (+) and a 5
```

If they are _not_ supposed to end a statement, then all programs must be a single statement.
```js
// error, because it is read as 5 5
5
5
```

I was thinking that they should function like the semicolon in JS (since one of the requirements is the ECMA requirement), but if AEMI is present, then the automatic semicolon insertation rule is useless (since it always gets inserted) (or drop AEMI instead). 

Maybe this is the intended way?

**Comment by TodePond** [comment]
```java
// 5 and 5, not 10
5 // seen as the literal 5
+ 5 // seen as a unary plus (+) and a 5
```

equivalent to

```java
5 + 5!
```

<hr>

```java
// error, because it is read as 5 5
5
5
```

equivalent to
```java
5!
5!
```

**Comment by circle-gon** [comment]
Thanks for the response! So it should work the same as js' automatic semicolon insertation rule?

**Comment by TodePond** [comment]
> So it should work the same as the automatic semicolon insertation rule (effectively replacing AEMI)?

not necessarily. the way all dreamberd 'rewrite rules' work is that it tries to run your code as-is, and only implicits anything if an error would happen otherwise - doing the smallest change required in these circumstances. i have written the order of these changes in a few issues/discussions, but it isnt fully explicit, so you can pick your own order for now
